include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -696,6 +696,8 @@ typedef void (*OrtCustomJoinThreadFn)(OrtCustomThreadHandle ort_custom_thread_ha
 
 typedef OrtStatus*(ORT_API_CALL* RegisterCustomOpsFn)(OrtSessionOptions* options, const OrtApiBase* api);
 
+typedef void (*RunAsyncCallbackFn)(OrtValue**, size_t, OrtStatusPtr);
```

Comment:
We might need to add another argument, like void*or siize_t.
So customers consuming this API would have a way to correlate the output values and a certain external handle.

Reply:
Any callback should allow user data to be passed back to the user. So, we should have something like the following:
1. When the callback is registered with ORT, the user should be allowed to pass a void* as user data.
2. When the callback is invoked by ORT, ORT should pass this user data to the callback.
Hence the signature of the callback can be something like:
```typedef void (*RunAsyncCallbackFn)(void* userdata, OrtValue**, OrtStatusPtr);```

Reply:
Done.